datacapture/src/main/java/com/google/android/fhir/datacapture/QuestionnaireValidationErrorMessageDialogFragment.kt

```diff
@@ -51,17 +51,23 @@ internal class QuestionnaireValidationErrorMessageDialogFragment(
 
   override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
     isCancelable = false
-    return MaterialAlertDialogBuilder(requireContext())
-      .setView(onCreateCustomView())
-      .setPositiveButton(R.string.questionnaire_validation_error_fix_button_text) { dialog, _ ->
+    val currentDialog =
+      MaterialAlertDialogBuilder(requireContext()).setView(onCreateCustomView()).setPositiveButton(
+        R.string.questionnaire_validation_error_fix_button_text,
+      ) { dialog, _ ->
         setFragmentResult(RESULT_CALLBACK, bundleOf(RESULT_KEY to RESULT_VALUE_FIX))
         dialog?.dismiss()
       }
-      .setNegativeButton(R.string.questionnaire_validation_error_submit_button_text) { dialog, _ ->
+    if (arguments == null || requireArguments().getBoolean(EXTRA_SHOW_SUBMIT_ANYWAY_BUTTON, true)) {
```

Comment:
requireArguments() already checks for arguments nullability.
```suggestion
    if (requireArguments().getBoolean(EXTRA_SHOW_SUBMIT_ANYWAY_BUTTON, true)) {
```

Reply:
i think if we're checking `arguments == null` we might as well just write `arguements.get...`... the point of the `requireArguments` function is to really require the arguments not to be null...

Reply:
I tried to replace this with `arguments.get...` but getting a suggestion to make the call as **non-null asserted**:

<img width="758" alt="image (11)" src="https://github.com/user-attachments/assets/bf56dd5a-2ea9-41f0-872d-21c7689c0fce">

Getting further suggestions to use `requireArguments()` if I make it as non-null asserted

<img width="786" alt="Screenshot 2024-10-01 at 4 51 02 PM" src="https://github.com/user-attachments/assets/b6f2fd08-5227-4647-b979-b9df4d10e6d9">

Considering this, I'll keep it as is.